src/test/groovy/com/palantir/gradle/graal/GradleGraalPluginIntegrationSpec.groovy

```diff
@@ -32,27 +47,50 @@ class GradleGraalPluginIntegrationSpec extends IntegrationSpec {
                 }
             }
         '''
+    }
 
-        buildFile << '''
-            apply plugin: 'java'
+    def 'allows specifying different graal version'() {
+        setup:
+        buildFile << """
             apply plugin: 'com.palantir.graal'
 
             graal {
-               mainClass 'com.palantir.test.Main'
-               outputName 'hello-world'
+               graalVersion '1.0.0-rc3'
+               downloadBaseUrl '${fakeBaseUrl}'
             }
-        '''
+        """
+        server.enqueue(new MockResponse().setBody('<<tgz>>'));
+
+        when:
+        ExecutionResult result = runTasksSuccessfully('downloadGraalTooling', "-Dcom.palantir.graal.cache.dir=${cacheDir}")
+
+        then:
+        result.wasExecuted(':downloadGraalTooling')
+        result.wasUpToDate(':downloadGraalTooling') == false
+        result.wasSkipped(':downloadGraalTooling') == false
+
+        server.takeRequest().requestUrl.toString() =~ "http://localhost:${server.port}" +
+                "/oracle/graal/releases/download//vm-1.0.0-rc3/graalvm-ce-1.0.0-rc3-(macos|linux)-amd64.tar.gz"
+        file("cacheDir/1.0.0-rc3/graalvm-ce-1.0.0-rc3-amd64.tar.gz").text == '<<tgz>>'
+    }
+
+    def 'downloadGraalTooling behaves incrementally'() {
+        setup:
+        buildFile << """
+            apply plugin: 'com.palantir.graal'
+
+            graal {
+               downloadBaseUrl '${fakeBaseUrl}'
+            }
+        """
+        server.enqueue(new MockResponse().setBody('<<tgz>>'));
 
         when:
-        ExecutionResult result = runTasks('nativeImage')
-        // capture output from Gradle runs
-        println "Gradle Standard Out:\n" + result.standardOutput
-        println "Gradle Standard Error:\n" + result.standardError
-        File output = new File(getProjectDir(), "build/graal/hello-world");
+        ExecutionResult result1 = runTasksSuccessfully('downloadGraalTooling', "-Dcom.palantir.graal.cache.dir=${cacheDir}")
+        ExecutionResult result2 = runTasksSuccessfully('downloadGraalTooling', "-Dcom.palantir.graal.cache.dir=${cacheDir}")
```

Comment:
put this in gradle.properties in `setup()` once this is read from a project prop, no need to pass it every time like this